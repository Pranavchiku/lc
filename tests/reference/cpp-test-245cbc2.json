{
    "basename": "cpp-test-245cbc2",
    "cmd": "lc --no-color --show-cpp {infile} -extra-arg=\"\"",
    "infile": "tests/test.cpp",
    "infile_hash": "9e5ef4c37b7e9e1f13f0082ce65459c1e596430887537eb8fd656b77",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "cpp-test-245cbc2.stdout",
    "stdout_hash": "27108a8e13274f0b14756c71f802f468b65e40d511b3a72cc9e30df8",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}