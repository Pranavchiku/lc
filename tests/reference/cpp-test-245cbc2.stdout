#include <iostream>
#include <string>
#include <vector>
#include <cassert>
#include <cmath>
#include <complex>
#include <Kokkos_Core.hpp>
#include <lfortran_intrinsics.h>

template <typename T>
Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
{
    Kokkos::View<T*> r("r", v.size());
    for (size_t i=0; i < v.size(); i++) {
        r(i) = v[i];
    }
    return r;
}

// Forward declarations
int32_t f(int32_t &x);

// Implementations
int32_t f(int32_t &x)
{
    int32_t __return_var;
    int32_t result;
    result = x/42;
    __return_var = result;
    return __return_var;
}

